source/develop/release-management/features/external-network-provider.html.md

```diff
@@ -0,0 +1,525 @@
+		],
+                ports: [
+                    {
+                        "status": "DOWN",
+                        "name": "nic5",
+                        "device_owner": "oVirt",
+                        "id": "49ccb785-eadb-469d-8c33-e7bc87d37e4e",
+                        "network_id": "bf864bf3-81d8-438d-bf68-4b0c357309b3",
+                        "device_id": "5cc10431-0b25-41bd-941c-3a1aed8edd87",
+                        "mac_address": "00:1a:4a:16:01:59"
+                        ...
+                        <more to be added if required>
+                    },
+                    ...
+                ]
+            },
+            ...
+        ]
+}
+```
+
+## User work-flows
+
+The user flows in the UI will mostly match the flows of "OpenStack Networking"
+
+### External network provider administration
+
+#### Adding an external network provider
+
+The external provider will be added from "External Providers".
+The UI Panel for adding a new external network provider will look as follows:
+![](external_network_provider_add.png "fig:external_network_provider_add.png")
+This will add the provider to the list of external providers.
+
+###### Read only option
+When the read only option is chosen, all the configuration items related to this provider will be read-only.
+This includes:
+
+*   no subnetworks can be added or removed
+
+*   when an external network is deleted, the option to remove it from a provider will be disabled.
+
+Creating and deleting of ports must be allowed as it is required to create and distroy VM NICs.
+
+#### Importing networks from external provider into oVirt (UI)
+This opens a dialog where networks from the external provider can be imported.
+Each external network is identified by an "id" given by the external provider.
+
+Same as OpenStack Neutron Integration
+
+External network provider interactions:
+
+* get all network from the external providers (GET networks)
+
+#### Delete external network
+
+An external network is deleted (Networks -> Remove).
+Same as OpenStack Neutron Integration
+
+External network provider interactions:
+When the "Remove network(s) from the external provider(s) as well." ceckbox is checked:
+
+* delete network (DELETE networks)
+
+#### Subnets management (UI)
+Same as OpenStack Neutron Integration
+
+The oVirt "Networks" tab has a "Subnets" subtab (only available for external
+networks), which can be used to list/add/remove the subnets of an external network.
+
+External network provider interactions when adding a subnet (Networks -> Subnets tab -> New):
+
+* Get network (GET networks/<id>)
+* Add subnet (POST subnets)
+
+External network provider interactions when removing a subnet (Networks -> Subnets tab -> Remove):
+* Add subnet (DELETE subnets/<id>)
+
+Opening the subnets subtab causes a request for the list of subnets
+* Get subnets (GET subnets)
+
+#### Setup networks dialog (UI) - optional
+The setup network dialog could be used to modify external networks. Some of the
```

Comment:
As you state, the current statement here is *very* problematic. What I /really/ want is to state something along the lines"

External network providers may depend on prior host-level connectivity. For example, OpenStack Neutron needs to be configured to use a specific bond (or NIC) for its VM networks. The network partner may ship a network_setup vdsm hook to integrate this configuration into the setup network dialog. For example, a custom property named "underlay" may be added to a network, which is attached to a bond0. When applied to the host, the hook script would do the necessary configuration to Neutron plugin so the "underlay" network's bond is used.